kafka_consumer/tests/conftest.py

```diff
@@ -8,46 +8,75 @@
 import pytest
 from confluent_kafka.admin import AdminClient
 from confluent_kafka.cimpl import NewTopic
-from datadog_test_libs.utils.mock_dns import mock_local
 
 from datadog_checks.dev import WaitFor, docker_run
+from datadog_checks.dev._env import e2e_testing
 from datadog_checks.kafka_consumer import KafkaCheck
 
-from .common import DOCKER_IMAGE_PATH, HERE, HOST_IP, KAFKA_CONNECT_STR, LEGACY_CLIENT, TOPICS
+from .common import AUTHENTICATION, DOCKER_IMAGE_PATH, HERE, KAFKA_CONNECT_STR, LEGACY_CLIENT, TOPICS
 from .runners import KConsumer, Producer
 
-# Dummy TLS certs
-CERTIFICATE_DIR = os.path.join(os.path.dirname(__file__), 'certificate')
-cert = os.path.join(CERTIFICATE_DIR, 'cert.cert')
-private_key = os.path.join(CERTIFICATE_DIR, 'server.pem')
-
+CERTIFICATE_DIR = os.path.join(os.path.dirname(__file__), 'docker', 'ssl', 'certificate')
+ROOT_CERTIFICATE = os.path.join(CERTIFICATE_DIR, 'caroot.pem')
+CERTIFICATE = os.path.join(CERTIFICATE_DIR, 'cert.pem')
+PRIVATE_KEY = os.path.join(CERTIFICATE_DIR, 'key.pem')
+PRIVATE_KEY_PASSWORD = 'secret'
 
 if LEGACY_CLIENT:
     E2E_METADATA = {
         'custom_hosts': [('kafka1', '127.0.0.1'), ('kafka2', '127.0.0.1')],
         'start_commands': [
             'apt-get update',
             'apt-get install -y build-essential',
+            'bash /tmp/start_commands.sh',
+        ],
+        'docker_volumes': [
+            f'{HERE}/scripts/start_commands.sh:/tmp/start_commands.sh',
+            f'{HERE}/docker/ssl/certificate:/tmp/certificate',
         ],
     }
 else:
     E2E_METADATA = {
         'custom_hosts': [('kafka1', '127.0.0.1'), ('kafka2', '127.0.0.1')],
-        'docker_volumes': [f'{HERE}/scripts/start_commands.sh:/tmp/start_commands.sh'],
+        'docker_volumes': [
+            f'{HERE}/scripts/start_commands.sh:/tmp/start_commands.sh',
+            f'{HERE}/docker/ssl/certificate:/tmp/certificate',
+        ],
         'start_commands': ['bash /tmp/start_commands.sh'],
     }
 
-INSTANCE = {
-    'kafka_connect_str': KAFKA_CONNECT_STR,
-    'tags': ['optional:tag1'],
-    'consumer_groups': {'my_consumer': {'marvel': [0]}},
-    'broker_requests_batch_size': 1,
-    'use_legacy_client': LEGACY_CLIENT,
-}
+if AUTHENTICATION == "ssl":
+    INSTANCE = {
+        'kafka_connect_str': "localhost:9092",
+        'tags': ['optional:tag1'],
+        'consumer_groups': {'my_consumer': {'marvel': [0]}},
+        'broker_requests_batch_size': 1,
+        'security_protocol': 'SSL',
+        'tls_cert': CERTIFICATE,
+        'tls_private_key': PRIVATE_KEY,
+        'tls_private_key_password': PRIVATE_KEY_PASSWORD,
+        'tls_ca_cert': ROOT_CERTIFICATE,
```

Comment:
```suggestion
        'tls_cert': "/tmp/certificate/cert.pem" if e2e_testing() else CERTIFICATE,
        'tls_private_key': "/tmp/certificate/key.pem" if e2e_testing() else PRIVATE_KEY,
        'tls_private_key_password': PRIVATE_KEY_PASSWORD,
        'tls_ca_cert': "/tmp/certificate/caroot.pem" if e2e_testing() else ROOT_CERTIFICATE,
```

I think doing this would allow us to just have one `INSTANCE` rather than a `E2E_INSTANCE` and separate logic in `def kafka_instance()`.

Reply:
That's a good idea however I'm also using the INSTANCE to create the `AdminClient` and `Producer`/`Consumer` in the conditions so I need to keep a "local" instance. The `INSTANCE` is then used to run the check in unit/integrations tests and the `E2E_INSTANCE` for the e2e tests. `INSTANCE` is also used in these three cases to set up the environment since it runs locally